app/jobs/hesa/retrieve_collection_job.rb

```diff
@@ -0,0 +1,35 @@
+# frozen_string_literal: true
+
+module Hesa
+  class RetrieveCollectionJob < ApplicationJob
+    def perform
+      return unless FeatureService.enabled?(:sync_from_hesa)
+
+      request_time = Time.zone.now
+      xml_response = Hesa::Client.get(url: url)
```

Comment:
Do we know what a failure response might look like?

Just wondering about the case where we fail to retrieve the collection, create a `HesaCollectionRequest` regardless, and then accidentally skip a day because we're incrementing the `next_from_date`.

Reply:
Oh yes. That would never do. Let me sort that out.

Reply:
Just been discussing this with @jhenderson offline. We're going to give the `HesaCollectionRequest` a state and if there's a problem (like the error response that you suggested) we'll mark it as failed and ignore it when setting the retrieving the next_from_date. So... I've not changed anything here other than the comment but we'll address it in subsequent PRs.

Reply:
Great news!